ext/phar/phar_object.c

```diff
@@ -2837,26 +2839,28 @@ PHP_METHOD(Phar, setStub)
 	size_t stub_len;
 	zend_long len = -1;
 	php_stream *stream;
-	PHAR_ARCHIVE_OBJECT();
 
-	if (PHAR_G(readonly) && !phar_obj->archive->is_data) {
-		zend_throw_exception_ex(spl_ce_UnexpectedValueException, 0,
-			"Cannot change stub, phar is read-only");
-		RETURN_THROWS();
-	}
+	if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS(), "r|l", &zstub, &len) == SUCCESS) {
```

Comment:
I'd either just leave this be, or refactor it the other way around (handle zpp specific parts first and then have a common codepath using the result).

Reply:
As I'm not that motivated to work on ext/phar, I'll keep this as-is. :)